presto-main/src/main/java/com/facebook/presto/metadata/MetadataManager.java

```diff
@@ -836,7 +836,7 @@ public void finishDelete(Session session, TableHandle tableHandle, Collection<Sl
             ConnectorViewDefinition view = views.get(viewName.asSchemaTableName());
             if (view != null) {
                 ViewDefinition definition = deserializeView(view.getViewData());
-                if (view.getOwner().isPresent()) {
+                if (view.getOwner().isPresent() && !definition.isRunAsInvoker()) {
```

Comment:
how could a view have an owner if runAsInvoker is also set?

Reply:
This is `ConnectorViewDefinition` which comes from the connector, not the serialized view. Connectors will set this unconditionally (they have no knowledge of the view security model). For example, in Hive, this is always set to the table owner returned by the metastore.

Reply:
Ah, I think I understand now.  Let me now if this is correct.
Viewdata doesn't contain information about the owner (for legacy views only?), so it's passed only with the connector view information.  Therefore we need some other field to say whether we want to get the view owner from the connector or not, and that's why we you added the "runAsInvoker" boolean.

Reply:
Yes, exactly